pkg/addons/client_test.go

```diff
@@ -12,25 +12,38 @@ func TestNeedsUpdate(t *testing.T) {
 		existing *unstructured.Unstructured
 		updated  *unstructured.Unstructured
 		exp      bool
+		diff     bool
 	}{
 		{
 			name:     "no need for upgrade",
 			existing: getObjectFromFile("testdata/service1.yaml"),
 			updated:  getObjectFromFile("testdata/service2.yaml"),
-			exp:      false,
+			exp:      false, // no upgrade expected
+			diff:     false, // no diff expected
 		},
 		{
 			name:     "needs upgrade",
 			existing: getObjectFromFile("testdata/service1.yaml"),
 			updated:  getObjectFromFile("testdata/service3.yaml"),
-			exp:      true,
+			exp:      true, // upgrade expected
+			diff:     true, // diff expected
+		},
+		{
+			name:     "secret diff omitted",
+			existing: getObjectFromFile("testdata/secret2.yaml"),
+			updated:  getObjectFromFile("testdata/secret3.yaml"),
+			exp:      true,  // upgrade expected
+			diff:     false, // no diff expected
 		},
 	}
 
 	for _, test := range tests {
 		if got := needsUpdate(test.existing, test.updated); got != test.exp {
 			t.Errorf("%s: expected update %t, got %t", test.name, test.exp, got)
 		}
+		if got := printDiff(test.existing, test.updated); got != test.diff {
```

Comment:
This unit test is specifically about needsUpdate (`TestNeedsUpdate`). You will have to create your own test function (`TestPrintDiff`).